lib/broadway/processor.ex

```diff
@@ -46,8 +46,9 @@ defmodule Broadway.Processor do
 
   @impl true
   def handle_events(messages, _from, state) do
+    start_time = System.monotonic_time()
     metadata = %{name: state.name, messages: messages}
-    measurements = %{time: System.monotonic_time()}
+    measurements = %{time: start_time}
```

Comment:
Wouldn't you want to have this telemetry start inside of `handle_messages/4` so that you get telemetry info for each message?

Reply:
For now we are going with a "macro" approach, we can always revisit it if folks need finer grained controls.

Reply:
Sounds good to me. Thanks for the clarification :)